Sources/Kevin/Core/Custom/KevinTapableText/KevinTapableText.swift

```diff
@@ -0,0 +1,212 @@
+    }
+
+    override var textAlignment: NSTextAlignment {
+        didSet { updateTextStorage()}
+    }
+
+    override var numberOfLines: Int {
+        didSet { textContainer.maximumNumberOfLines = numberOfLines }
+    }
+
+    override var intrinsicContentSize: CGSize {
+        guard let text = text, !text.isEmpty else {
+            return .zero
+        }
+
+        textContainer.size = CGSize(width: self.preferredMaxLayoutWidth, height: CGFloat.greatestFiniteMagnitude)
+        let size = layoutManager.usedRect(for: textContainer)
+        return CGSize(width: ceil(size.width), height: ceil(size.height))
+    }
+
+    // MARK: Private properties
+
+    private let layoutManager = NSLayoutManager()
+    private let textContainer = NSTextContainer(size: CGSize.zero)
+    private var textStorage = NSTextStorage()
+    private var lockUpdating = true
+    private var heightCorrection: CGFloat = 0
+    private var activeElements = [ElementTuple]()
+
+    typealias ElementTuple = (range: NSRange, element: KevinTapableTextLink)
+
+    override public init(frame: CGRect) {
+        super.init(frame: frame)
+        lockUpdating = false
+        setupLabel()
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        lockUpdating = false
+        setupLabel()
+    }
+    
+    override func drawText(in rect: CGRect) {
+        let range = NSRange(location: 0, length: textStorage.length)
+        
+        textContainer.size = rect.size
+        let usedRect = layoutManager.usedRect(for: textContainer)
+        heightCorrection = (rect.height - usedRect.height)/2
+        let glyphOriginY = heightCorrection > 0 ? rect.origin.y + heightCorrection : rect.origin.y
+        let newOrigin = CGPoint(x: rect.origin.x, y: glyphOriginY)
+        
+        layoutManager.drawBackground(forGlyphRange: range, at: newOrigin)
+        layoutManager.drawGlyphs(forGlyphRange: range, at: newOrigin)
+    }
+
+    // MARK: Label setup
+
+    private func setupLabel() {
+        textStorage.addLayoutManager(layoutManager)
+        layoutManager.addTextContainer(textContainer)
+        textContainer.lineFragmentPadding = 0
+        textContainer.lineBreakMode = lineBreakMode
+        textContainer.maximumNumberOfLines = numberOfLines
+        isUserInteractionEnabled = true
+    }
+    
+    private func updateTextStorage() {
+        if lockUpdating { return }
+
+        activeElements.removeAll()
+        guard let attributedText = attributedText, attributedText.length > 0 else {
+            textStorage.setAttributedString(NSAttributedString())
+            setNeedsDisplay()
+            return
+        }
+
+        let mutableAttributedString = NSMutableAttributedString(attributedString: attributedText)
+        setupParagraphStyle(mutableAttributedString)
+        setupTapableLinks(for: mutableAttributedString)
+        textStorage.setAttributedString(mutableAttributedString)
+        lockUpdating = true
```

Comment:
this part feels a little flaky to me. As I understand it, this is meant to prevent infinite recursion when text value is being set. 

Maybe instead to use private text value. What I mean text can be computed property. And setting _text would not call updateTextStorage() 

```    
    private var _text: String?
    override var text: String? {
        set {
             _text = newValue
            updateTextStorage()
        }
        get {
            _text
        }
    }
```